rclpy/rclpy/topic_endpoint_info.py

```diff
@@ -167,6 +167,7 @@ def __str__(self):
             f'GID: {gid}',
             'QoS profile:',
             f'  Reliability: {self.qos_profile.reliability.name}',
+            f'  History (Depth): {self.qos_profile.history.name} ({self.qos_profile.depth})',
```

Comment:
The issue is that the "get actual qos" methods aren't populating the history kind and history depth in some rmw implementations, so in those cases printing them just adds confusion.

Reply:
Is that not part of the rmw base spec? And if so, seems still worth making the change to expose it more and upstream reports to the rmw packages.

Reply:
@ivanpauno thanks for the info. Do we have dedicated issue about `get actual qos methods aren't populating the history kind and history depth in some rmw implementations`?

Reply:
Sorry, the issue is not in get_actual_qos(), but in the get_topic_endpoint_info() methods!

Reply:
@ivanpauno appreciate for informative comments, i will check them out! thanks!

Reply:
after all, only rmw_cyclonedds has the correct QoS history and depth via GraphCache. (as @ivanpauno mentioned.)

details are below.

https://github.com/ros2/rmw/blob/35fc6ab8fad4db90eb55db9d1ecf50dc1aa3638d/rmw/include/rmw/get_topic_endpoint_info.h#L39-L40

https://github.com/ros2/rmw_fastrtps/blob/32ce21d2d0250a46b9eeda9df1cdcd76503a5823/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp#L150

https://github.com/ros2/rmw_fastrtps/blob/32ce21d2d0250a46b9eeda9df1cdcd76503a5823/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/qos.hpp#L125-L126

-> rmw_fastrtps does not fill in history and depth, this can be confirmed with talker sample. (rmw_connextdds does have the same behavior.)

https://github.com/ros2/demos/blob/7f8cd3aac2a179288bf02a412432f43c4d451355/demo_nodes_cpp/src/topics/talker.cpp#L56

```
root@tomoyafujita:~/ros2_ws/colcon_ws# RMW_IMPLEMENTATION=rmw_fastrtps_cpp ros2 topic info -v /chatter
Type: std_msgs/msg/String

Publisher count: 1

Node name: talker
Node namespace: /
Topic type: std_msgs/msg/String
Endpoint type: PUBLISHER
GID: 01.0f.64.e2.6e.26.99.24.01.00.00.00.00.00.12.03.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): KEEP_LAST (1) !!! HERE, this is supposed to be 7 !!!
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Subscription count: 0
```